sql/core/src/test/scala/org/apache/spark/sql/ShowCreateTableSuite.scala

```diff
@@ -182,6 +183,27 @@ abstract class ShowCreateTableSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("SPARK-36012: Add NULL flag when SHOW CREATE TABLE") {
+    val t = "SPARK_36012"
+    withTable(t) {
+      sql(
+        s"""
+           |CREATE TABLE $t (
+           |  a bigint NOT NULL,
+           |  b bigint
+           |)
+           |USING ${classOf[SimpleInsertSource].getName}
+        """.stripMargin)
+      val showDDL = sql(s"SHOW CREATE TABLE $t").head().getString(0)
+        .split("\n").map(_.trim)
+      assert(showDDL === Array(
+        "CREATE TABLE `default`.`SPARK_36012` (",
+        "`a` BIGINT NOT NULL,",
+        "`b` BIGINT)",
+        "USING org.apache.spark.sql.sources.SimpleInsertSource"))
```

Comment:
super nit and you don't have to change: I think you can modify `getShowDDL` a bit and reuse it, and make this code structure the same as `test("SPARK-24911: keep quotes for nested fields")`

Reply:
@huaxingao Thank you for your code review. By the way, change it.

Reply:
I really don't like this function `getShowDDL`. It just returns part of the `DDL` data instead of the complete `DDL` data. But it is named `getShowDDL`